### PR TITLE
RBAC: Rewrite rbac annotations test

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
Rewrite RBAC API tests for annotations to no longer use mocked access control

### CODE REVIEW

**Review by IevaVasiljeva — APPROVED** [review]
Looks good overall (and much shorter!), but I think some of the removed tests could be added back in and some test cases are missing from the new version.

File: pkg/api/annotations_test.go [hunk 1]
```diff
@@ -386,587 +388,228 @@ func deleteAnnotationsScenario(t *testing.T, desc string, url string, routePatte
...
-				RegisterScopeAttributeResolver(AnnotationTypeScopeResolver(sc.hs.annotationsRepo))
-			setAccessControlPermissions(sc.acmock, tt.args.permissions, sc.initCtx.OrgID)
-
-			r := callAPI(sc.server, tt.args.method, tt.args.url, tt.args.body, t)
-			assert.Equalf(t, tt.want, r.Code, "Annotations API(%v)", tt.args.url)
-		})
-	}
-}
-
-func TestService_AnnotationTypeScopeResolver(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (kalleep) [reply pkg/api/annotations_test.go:754]: ops that it was a mistake to remove this one

File: pkg/api/annotations_test.go [hunk 2]
```diff
@@ -386,587 +388,228 @@ func deleteAnnotationsScenario(t *testing.T, desc string, url string, routePatte
...
-			setAccessControlPermissions(sc.acmock, tt.args.permissions, sc.initCtx.OrgID)
-			dashboardAnnotation := &annotations.Item{Id: 1, DashboardId: 1}
-			organizationAnnotation := &annotations.Item{Id: 2, DashboardId: 0}
-
-			_ = sc.hs.annotationsRepo.Save(context.Background(), dashboardAnnotation)
-			_ = sc.hs.annotationsRepo.Save(context.Background(), organizationAnnotation)
+			server := SetupAPITestServer(t, func(hs *HTTPServer) {
+				hs.Cfg = setting.NewCfg()
+				repo := annotationstest.NewFakeAnnotationsRepo()
+				_ = repo.Save(context.Background(), &annotations.Item{Id: 2, DashboardId: 1})
```
[the remark this answers is not in the document]
Comment (kalleep) [reply pkg/api/annotations_test.go:598]: The default return value of the fake is &annotations.Item{Id: 1, DashboardId: 0} if no other is found. But can can add it to make it more clear and if the fake changes this does not break

[the remark this answers is not in the document]
Comment (IevaVasiljeva) [reply pkg/api/annotations_test.go:599]: Gotcha, that makes sense :)

**Review by IevaVasiljeva — APPROVED** [review]

**Pull request merged**